latex/epoch.tex

```diff
@@ -1,54 +1,614 @@
+        \var{pc}\\
+      \end{array}
+      \vdash
+      \left(
+        \begin{array}{r}
+          \var{utxoSt} \\
+          \var{dstate} \\
+          \var{stpools} \\
+          \var{pparams} \\
+          \var{retiring}
+        \end{array}
+      \right)
+      \trans{poolclean}{}
+      \left(
+        \begin{array}{rcl}
+          \var{utxoSt'} \\
+          \var{dstate'} \\
+          \var{retired} & \subtractdom & \var{stpools} \\
+          \var{retired} & \subtractdom & \var{pparams} \\
+          \var{retired} & \subtractdom & \var{retiring} \\
+        \end{array}
+      \right)
+    }
+  \end{equation}
+  \caption{Pool Clean Inference Rule}
+  \label{fig:rules:pool-clean}
+\end{figure}
+
+%%
+%% Figure - New Proto Consts Defs
+%%
+\begin{figure}
+  \emph{New Proto Consts environment}
+  \begin{equation*}
+    \NewProtoConstsEnv =
+    \left(
+      \begin{array}{r@{~\in~}lr}
+        \var{slot} & \Slot & \text{current slot}\\
+        \var{pcOld} & \PrtclConsts & \text{old protocol constants}\\
+        \var{pcNew} & \PrtclConsts & \text{new protocol constants}\\
+        \var{dstate} & \DState & \text{delegation state}\\
+        \var{pstate} & \PState & \text{pool state}\\
+      \end{array}
+    \right)
+  \end{equation*}
+  %
+  \emph{New Proto Consts States}
+  \begin{equation*}
+    \NewProtoConstsState =
+    \left(
+      \begin{array}{r@{~\in~}lr}
+        \var{utxoSt} & \UTxOState & \text{utxo state}\\
+        \var{accnt} & \Accnt & \text{accounting}\\
+      \end{array}
+    \right)
+  \end{equation*}
+  %
+  \emph{New Proto Consts transitions}
+  \begin{equation*}
+    \_ \vdash
+    \var{\_} \trans{newpc}{} \var{\_}
+    \subseteq \powerset (\NewProtoConstsEnv \times
+    \NewProtoConstsState \times \NewProtoConstsState)
+  \end{equation*}
+  %
+  \caption{New Proto Consts transition-system types}
+  \label{fig:ts-types:new-proto-consts}
+\end{figure}
+
+%%
+%% Figure - New Proto Consts Rule
+%%
+\begin{figure}
+  \begin{equation}\label{eq:new-pc}
+    \inference[New-Proto-Consts]
+    {
+      \var{oblgOld} = \obligation{pcOld}{stkeys}{stpools}{slot} \\
+      \var{oblgNew} = \obligation{pcNew}{stkeys}{stpools}{slot} \\
+      ~\\
+      \var{reserves} + \var{oblgOld} \geq \var{oblgNew}\\
+      \var{diff} = \var{oblgOld} - \var{oblgNew} \\
+      ~\\
+      \var{utxoSt'} =
+      \left(
+        {
+          \begin{array}{r}
+            \var{utxo} \\
+            \var{oblgNew} \\
+            \var{fees} \\
+            \var{wdrls} \\
+          \end{array}
+        }
+      \right)
+      &
+      \var{accnt'} =
+      \left(
+        {
+          \begin{array}{r}
+            \var{treasury} \\
+            \var{reserves} + \var{diff} \\
+            \var{rewardPool} \\
+            \var{rewards} \\
+          \end{array}
+        }
+      \right)
+    }
+    {
+      \begin{array}{l}
+        \var{slot}\\
+        \var{pc}\\
+      \end{array}
+      \vdash
+      \left(
+        \begin{array}{r}
```

Comment:
This is strange, the variables `utxoSt` and `accnt` aren't used. Aren't we missing something here?

Reply:
This is similar to the `stkeys` and `delegations` comments above, it has to do with packing and upacking data for the `EPOCH` rule.